src/runtime-rs/crates/resource/src/network/endpoint/ipvlan_endpoint.rs

```diff
@@ -4,37 +4,54 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-use std::io::{self, Error};
+use std::{
+    io::{self, Error},
+    sync::Arc,
+};
 
-use super::endpoint_persist::{EndpointState, IpVlanEndpointState};
 use anyhow::{Context, Result};
 use async_trait::async_trait;
-use hypervisor::device::DeviceType;
-use hypervisor::NetworkDevice;
+use tokio::sync::RwLock;
 
-use super::Endpoint;
-use crate::network::network_model::TC_FILTER_NET_MODEL_STR;
-use crate::network::{utils, NetworkPair};
-use hypervisor::{device::driver::NetworkConfig, Hypervisor};
+use hypervisor::{
+    device::{
+        device_manager::{do_handle_device, DeviceManager},
+        driver::NetworkConfig,
+        DeviceConfig, DeviceType,
+    },
+    Hypervisor, NetworkDevice,
+};
+
+use super::{
+    endpoint_persist::{EndpointState, IpVlanEndpointState},
+    Endpoint,
+};
+use crate::network::{network_model::TC_FILTER_NET_MODEL_STR, utils, NetworkPair};
 
 // IPVlanEndpoint is the endpoint bridged to VM
 #[derive(Debug)]
 pub struct IPVlanEndpoint {
     pub(crate) net_pair: NetworkPair,
+    pub(crate) d: Arc<RwLock<DeviceManager>>,
```

Comment:
ditto

Reply:
By the way, is it possible to decouple the device manager's reference with the device? Or it is just the practice of using this.